OracleHook: Add Oracle SQL Support

yoziru opened the pull request:
Basic set-up to add Oracle SQL support to Airflow as a Hook + additional code to make sure insert_rows, Data Profiling and Charts work

yoziru committed:
Create oracle_hook.py

yoziru committed:
Update __init__.py

Add oracle_hook integration

yoziru committed:
Update oracle_hook.py

add dependencies

yoziru committed:
Update utils.py

add support for Data Profiling with Oracle SQL

yoziru committed:
Update models.py

add Oracle SQL support through the OracleHook

yoziru committed:
Update oracle_hook.py

add comments, remove unnecessary code

mistercrunch commented on airflow/hooks/oracle_hook.py:
```diff
@@ -0,0 +1,79 @@
+        conn = self.get_connection(self.oracle_conn_id)
+        try:
+            conn = cx_Oracle.connect(conn.login, conn.password, conn.host)
+        except:
```
Why is this in an except clause? Seems like it should be in an `if`.

You may want to clarify the connection options (especially the extra parameters) in the docstring

mistercrunch commented on airflow/hooks/oracle_hook.py:
```diff
@@ -0,0 +1,79 @@
+            conn = cx_Oracle.connect(conn.login, conn.password, dsn=dsn)
+        return conn
+
+    def get_pandas_df(self, sql, parameters = None):
```
I'm not sure if the lower case change is necessary, but if you think it is you should call `super` and then alter the column list to avoid duplicating the rest of the code.

Nit: no spaces around `=` in function header

mistercrunch commented on airflow/hooks/oracle_hook.py:
```diff
@@ -0,0 +1,79 @@
+        conn.close()
+        return df
+
+    def insert_rows(self, table, rows, target_fields = None, commit_every = 1000):
```
Can you highlight the difference from the parent class? Looks very similar at a glance

yoziru committed:
Update setup.py

Add cx_Oracle requirements to extras_require for the OracleHook

yoziru committed:
Update oracle_hook.py

Rework code, add additional dependencies and additional docs
Remove unnecessary code
